File: mobile/src/entities/medication/lib/dayCompleteFeed.ts
/** daily_logs.message — 하루 약 전부 복용 피드 마커 (유저 노출 X) */
export const DAY_COMPLETE_FEED_MARKER = 'day_complete' as const;

export function isDayCompleteFeedLog(log: {
  status: string | null;
  medicationId: number | null;
  message: string | null;
  condition?: string | null;
}): boolean {
  return (
    log.status === 'TAKEN' &&
    log.medicationId == null &&
    log.message === DAY_COMPLETE_FEED_MARKER &&
    !log.condition
  );
}

File: android-app/core/src/main/kotlin/com/jinlabs/yakok/core/med/DayCompleteFeed.kt
package com.jinlabs.yakok.core.med

/** daily_logs.message — 하루 약 전부 복용 피드 마커 (유저 노출 X) */
object DayCompleteFeed {
    const val Marker = "day_complete"

    fun isFeedLog(log: DailyLog): Boolean =
        log.status == LogStatus.Taken &&
            log.medicationId == null &&
            log.message == Marker &&
            log.condition == null
}
